### PR TITLE
TST/ENH: More thorough ALL tests for date and timestamp arithmetics

### PR DESCRIPTION
**Pull request by kszucs**
Implementations: clickhouse, impala
Support for mysql blob type.
`Schema.apply_to`uses numpy to apply timedelta dtypes.

### CODE REVIEW

File: ibis/impala/compiler.py [hunk 1]
```diff
@@ -615,18 +615,49 @@ def _table_array_view(translator, expr):
...
+
+    return func(translator, expr)
+
+
+def _timestamp_subtract(translator, expr):
+    op = expr.op()
+    left, right = op.args
+
+    if isinstance(right, ir.TimestampValue):
+        return 'unix_timestamp({}) - unix_timestamp({})'.format(
```
Comment (cpcloud) [inline ibis/impala/compiler.py]: Does this pass bigquery tests?

Comment (kszucs) [reply ibis/impala/compiler.py]: `TimestampAdd`, `TimestampSubtract`, `DateAdd`, `DateSubtract` are disabled in bigquery, see [here](https://github.com/ibis-project/ibis/pull/1352/files#diff-57418510fe1d988e5cb1614da6247204R267)

**Comment by kszucs** [comment]
@cpcloud issues resolved

**Comment by cpcloud** [comment]
@kszucs can you rebase?

File: ibis/bigquery/compiler.py [hunk 2]
```diff
@@ -265,6 +265,12 @@ def _timestamp_op(func, units):
     def _formatter(translator, expr):
         op = expr.op()
         arg, offset = op.args
+        if not isinstance(offset, ir.IntervalValue):
+            raise com.UnsupportedOperationError(
+                'Binary operations between two timestamps or dates '
+                'are not yet supported in bigquery backend'
+            )
```
Comment (kszucs) [inline ibis/bigquery/compiler.py]: Disable just for short term. 

Should we create [separate operations](https://github.com/ibis-project/ibis/blob/master/ibis/expr/operations.py#L2560) for 
- `timestamp - timestamp` 
- `timestamp - interval`
- `date - date`
- `date - interval`

That might be more convenient.

Comment (cpcloud) [reply ibis/bigquery/compiler.py]: Yep, those should be separate

Comment (cpcloud) [reply ibis/bigquery/compiler.py]: We can choose which one to construct inside the functions in `api.py`

Comment (cpcloud) [reply ibis/bigquery/compiler.py]: How about:
* `TimestampDiff`
* `TimestampSub`
* `DateDiff`
* `DateSub`

Comment (kszucs) [reply ibis/bigquery/compiler.py]: I was considering diff and sub too.

Comment (cpcloud) [reply ibis/bigquery/compiler.py]: @kszucs can you make those changes in another PR?

Comment (kszucs) [reply ibis/bigquery/compiler.py]: @cpcloud sure

Comment (cpcloud) [reply ibis/bigquery/compiler.py]: Great!

**Pull request closed without merging**